vm_test.go

```diff
@@ -561,6 +561,60 @@ func TestStashMemUsage(t *testing.T) {
 	}
 }
 
+func TestTmpValuesMemUsage(t *testing.T) {
+	tests := []struct {
+		name        string
+		val         []Value
+		memLimit    uint64
+		expectedMem uint64
+		errExpected error
+	}{
+		{
+			name:        "should account for no memory usage given an empty tmpValues",
+			val:         []Value{},
+			memLimit:    100,
+			expectedMem: 0,
+			errExpected: nil,
+		},
+		{
+			name:        "should account for no memory usage given a tmpValues with nil",
+			val:         []Value{nil},
+			memLimit:    100,
+			expectedMem: 0,
+			errExpected: nil,
+		},
+		{
+			name:        "should account for each value given a non-empty tmpValues",
+			val:         []Value{valueInt(99)},
+			memLimit:    100,
+			expectedMem: SizeInt,
+			errExpected: nil,
+		},
+		{
+			name:        "should exit early given tmpValues over the memory limit",
+			val:         []Value{valueInt(99), valueInt(99), valueInt(99), valueInt(99)},
+			memLimit:    0,
+			expectedMem: SizeInt,
```

Comment:
[nit] thoughts on adding a similar test without the limit set to 0 to make sure that we're iterating over all values.

Reply:
I updated the test above to account for multiple values, does that work for you?

Reply:
💯 perfect!